### PR TITLE
Feature/feature user api

### PR DESCRIPTION
**Pull request by marclll**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/controller/usermanager/UserRestController.java  80
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5247081&bid=17196374)

File: src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    @MockBean
+    private UserService service;
+
+    @BeforeAll
+    public static void setUp() {
+        urlBase = "/users";
+        username = "PotatoMaster";
+        userEntity = new UserEntity();
+        jsonEmail = "{\n"
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java:45]: Codacy Issue found: [The String "{\n" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5247081)

File: src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    @Test
+    public void getUserDataShouldReturnUserDataAndStatusOk() throws Exception {
+        willReturn(userEntity).given(service).getUserData(username);
+        mockMvc.perform(get(urlBase + "/" + username))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    public void updateEmailShouldReturnStatusOk() throws Exception {
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java:68]: Codacy Issue found: [Throwing 'Exception' is not allowed.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5247081)

File: src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        jsonEmail = "{\n"
+            + "  \"newEmail\": \"xavi@nanako.com\"\n"
+            + "} ";
+        jsonPassword = "{\n"
+            + "  \"newPasswordl\": \"hiufdahs\"\n"
+            + "} ";
+    }
+
+    @Test
+    public void deleteAccountShouldReturnStatusOk() throws Exception {
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java:54]: Codacy Issue found: [Throwing 'Exception' is not allowed.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5247081)

File: src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        willDoNothing().given(service).deleteById(anyString());
+        mockMvc.perform(delete(urlBase + "/" + username + "/delete"))
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java:56]: Codacy Issue found: [The String "/" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5247081)

File: src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    private UserService service;
+
+    @BeforeAll
+    public static void setUp() {
+        urlBase = "/users";
+        username = "PotatoMaster";
+        userEntity = new UserEntity();
+        jsonEmail = "{\n"
+            + "  \"newEmail\": \"xavi@nanako.com\"\n"
+            + "} ";
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java:47]: Codacy Issue found: [The String "} " appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5247081)

File: src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+    @Test
+    public void deleteAccountShouldReturnStatusOk() throws Exception {
+        willDoNothing().given(service).deleteById(anyString());
+        mockMvc.perform(delete(urlBase + "/" + username + "/delete"))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    public void getUserDataShouldReturnUserDataAndStatusOk() throws Exception {
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java:61]: Codacy Issue found: [Throwing 'Exception' is not allowed.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5247081)

File: src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    public void updateEmailShouldReturnStatusOk() throws Exception {
+        willDoNothing().given(service).updateEmail(anyString(), anyString());
+        mockMvc.perform(put(urlBase + "/" + username + "/update/email")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(jsonEmail))
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    public void updatePasswordShouldReturnStatusOk() throws Exception {
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/controller/usermanager/UserRestControllerTest.java:77]: Codacy Issue found: [Throwing 'Exception' is not allowed.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5247081)

**Review by xavier-campos-diaz — APPROVED** [review]

**Review by AlbertPG — APPROVED** [review]
I see no problem at all

**Pull request merged**